### PR TITLE
Common/Monorepo: Remove NetworkId

### PR DESCRIPTION
**Pull request by holgerd77**
This PR removes `networkId` from Common and upstream usages and configuration files. Client is keeping for now as a CLI option (but deprecated) for a smoother transition on things like Docker and stuff.

Ready for review! 🙂

### CODE REVIEW

File: packages/client/bin/cli.ts [hunk 1]
```diff
@@ -923,7 +929,7 @@ async function run() {
   // TODO sharding: Just initialize kzg library now, in future it can be optimized to be
   // loaded and initialized on the sharding hardfork activation
   // Give network id precedence over network name
```
Comment (gabrocheleau) [inline packages/client/bin/cli.ts:931]:
```suggestion
  // Give chainId priority over networkId
  // Give networkId precedence over network name
```

File: packages/common/test/customChains.spec.ts [hunk 2]
```diff
@@ -63,12 +61,12 @@ describe('[Common]: Custom chains', () => {
 
   it('custom() -> behavior', () => {
     let common = createCustomCommon({ chainId: 123 })
-    assert.deepEqual(common.networkId(), BigInt(1), 'should default to mainnet base chain')
+    assert.equal(common.consensusAlgorithm(), 'casper', 'should default to mainnet base chain')
```
Comment (gabrocheleau) [inline packages/common/test/customChains.spec.ts:64]: Is this right? Why are we not checking chainID?

Comment (holgerd77) [reply packages/common/test/customChains.spec.ts:64]: Yes, that's correct, the goal of this test is to *not* check for the value passed (`chainId`), but to check that the other values remain the mainnet-specific ones.

**Comment by holgerd77** [comment]
Have addressed, ready for re-review!

File: packages/client/bin/cli.ts [hunk 3]
```diff
@@ -85,8 +85,14 @@ const args: ClientOpts = yargs
...
   })
+  .option('chainId', {
+    describe: 'Chain ID',
+    choices: networks.map((n) => parseInt(n[0])).filter((el) => !isNaN(el)),
+    default: undefined,
+    conflicts: ['customChain', 'customGenesisState', 'gethGenesis'], // Disallows custom chain data and chainId
+  })
   .option('networkId', {
-    describe: 'Network ID',
+    describe: 'Network ID (deprecated, use chainId instead)',
```
Comment (ScottyPoi) [inline packages/client/bin/cli.ts:95]:
```suggestion
    describe: 'Network ID',
    deprecated: true,
    deprecate: 'use --chainId instead',
```

**Review by jochem-brouwer — APPROVED** [review]
LGTM!

**Pull request merged**